cirq/testing/random_circuit.py

```diff
@@ -72,15 +76,23 @@ def random_circuit(qubits: Union[Sequence[ops.Qid], int],
     if n_qubits < 1:
         raise ValueError('At least one qubit must be specified.')
 
+    if isinstance(random_state, np.random.RandomState):
+        prng = random_state
+    else:
+        prng = np.random.RandomState(random_state)
+
     moments: List[ops.Moment] = []
     for _ in range(n_moments):
         operations = []
-        free_qubits = set(q for q in qubits)
+        free_qubits = set(qubits)
         while len(free_qubits) >= max_arity:
-            gate, arity = choice(tuple(gate_domain.items()))
-            op_qubits = sample(free_qubits, arity)
+            gate, arity = tuple(gate_domain.items())[prng.randint(
```

Comment:
Seems wasteful to keep recomputing the tuple and length, but probably doesn't matter.

Reply:
Done.